torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -477,17 +477,14 @@ void TensorPipeAgent::sendCompletedResponseMessage(
           << " is sending response to request #" << messageId << " to "
           << pipe->getRemoteName();
 
-  const c10::optional<utils::FutureError> error =
-      futureResponseMessage->error();
-  Message&& responseMessage = std::move(*futureResponseMessage).moveValue();
-  responseMessage.setId(messageId);
-  if (!error) {
+  if (!futureResponseMessage->hasError()) {
+    Message&& responseMessage =
+        std::move(*futureResponseMessage->value().toCustomClass<Message>());
```

Comment:
I wonder: does it make sense to add a moveValue() method to ivalue::Future? For cases like this where we don't care about the future anymore and just want its content?

(Personally I think the "idiomatic" way would be to have a &&-qualified value() method but I've seen that in PyTorch that's not always the convention...)

Reply:
Sure, I can add that. cc @wanchaol if there are concerns for adding `value() &&`.